torch/autograd/_functions/tensor.py

```diff
@@ -591,4 +591,22 @@ def backward(self, grad_output):
         return grad_input
 
 
+class Cumsum(Function):
+
+    def __init__(self, dim):
+        super(Cumsum, self).__init__()
+        self.dim = dim
+
+    def forward(self, input):
+        return torch.cumsum(input, dim=self.dim)
+
+    def backward(self, grad_output):
+        grad_sum = torch.sum(grad_output, dim=self.dim)
+
+        grad_input = torch.cumsum(-grad_output, dim=self.dim)
+        grad_input += grad_sum.expand_as(grad_input)
```

Comment:
Isn't it possible to replace this line with `grad_input -= grad_input[-1]` and remove the line summing over the `grad_output` ?